production/helm/loki/src/helm-test/canary_test.go

```diff
@@ -0,0 +1,102 @@
+package test
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+	"github.com/stretchr/testify/require"
+)
+
+func getEnv(key, fallback string) string {
+	if value, ok := os.LookupEnv(key); ok {
+		return value
+	}
+	return fallback
+}
+
+func testResult(result model.Value, query string, test func(model.SampleValue) bool, msg string) error {
+	if v, ok := result.(model.Vector); ok {
+		for _, s := range v {
+			fmt.Printf("%s => %v\n", query, s.Value)
+			if !test(s.Value) {
+				return errors.New(msg)
+			}
+		}
+
+		return nil
+	}
+
+	return fmt.Errorf("unexpected Prometheus result type: %v ", result.Type())
+}
+
+func TestCanary(t *testing.T) {
+	totalEntriesQuery := "sum(loki_canary_entries_total)"
+	totalEntriesMissingQuery := "sum(loki_canary_missing_entries_total)"
+
+	testContext := func(t *testing.T) v1.API {
```

Comment:
```suggestion
	newClient := func(t *testing.T) v1.API {
```

Actually, this method could be moved out of the test function.